sdk/storage/azblob/sas/account.go

```diff
@@ -29,15 +30,19 @@ type AccountSignatureValues struct {
 	Protocol      Protocol  `param:"spr"` // See the SASProtocol* constants
 	StartTime     time.Time `param:"st"`  // Not specified if IsZero
 	ExpiryTime    time.Time `param:"se"`  // Not specified if IsZero
-	Permissions   string    `param:"sp"`  // Create by initializing a AccountSASPermissions and then call String()
+	Permissions   string    `param:"sp"`  // Create by initializing AccountPermissions and then call String()
 	IPRange       IPRange   `param:"sip"`
-	ResourceTypes string    `param:"srt"` // Create by initializing AccountSASResourceTypes and then call String()
+	ResourceTypes string    `param:"srt"` // Create by initializing AccountResourceTypes and then call String()
 }
 
 // SignWithSharedKey uses an account's shared key credential to sign this signature values to produce
 // the proper SAS query parameters.
 func (v AccountSignatureValues) SignWithSharedKey(sharedKeyCredential *SharedKeyCredential) (QueryParameters, error) {
 	// https://docs.microsoft.com/en-us/rest/api/storageservices/Constructing-an-Account-SAS
+	if sharedKeyCredential == nil {
```

Comment:
The convention is to not perform `nil` checks on required parameters.

Reply:
`SignWithSharedKey` is an exposed method and can be directly called as `sas.SignWithSharedKey()`. So, I added this check since `nil` can be passed as a `*SharedKeyCredential` parameter.

Reply:
Understood.  However, having this panic due to a missing required parameter is ok and is the convention (if the panic would occur at some other point post this method call then checking for `nil` is ok).

Reply:
Done. Removed this check from queues package as well.